### PR TITLE
include both page title and section header in search results

### PR DESCRIPTION
**Pull request by bgdeutsch**
### What does this PR do?
render both the page title and section header (when applicable) in the page hierarchy of search results.

### Motivation
https://datadoghq.atlassian.net/browse/WEB-3507

### Preview
do some queries from the search results page, confirm both page title and section header are present in the page hierarchy.
i.e. `Documentation » Integrations » Postgres » Data Collected`

https://docs-staging.datadoghq.com/brian.deutsch/integrations-search-result-header/search/?s=postgres
https://docs-staging.datadoghq.com/brian.deutsch/integrations-search-result-header/search/?s=aws
https://docs-staging.datadoghq.com/brian.deutsch/integrations-search-result-header/search/?s=synthetics

there should be no console errors, and styles should be unaffected.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by kayayarai — APPROVED** [review]

**Review by hestonhoffman — APPROVED** [review]
LGTM! Let me know if you need a review from @DataDog/corpweb, or if it's good to merge

**Review by StefonSimmons — APPROVED** [review]

**Pull request merged**
